thirdparty/openwsman/openwsman.patch

```diff
@@ -79,3 +79,39 @@ diff -rupN orig/src/lib/wsman-soap.c patched/src/lib/wsman-soap.c
                          max_connections_per_thread = (* fptr)();
                  }
                  else{
+
+diff -rupN orig/src/lib/u/lock.c patched/src/lib/u/lock.c
+--- orig/src/lib/u/lock.c	2021-05-31 13:44:43.992941115 +0200
++++ patched/src/lib/u/lock.c	2021-05-31 12:00:21.972733061 +0200
+@@ -50,7 +50,7 @@
+ extern int pthread_mutexattr_settype(pthread_mutexattr_t *attr, int type);
+ #endif
+ 
+-#if defined (__SVR4) && defined (__sun)
++#if (defined (__SVR4) && defined (__sun)) || !defined(__GLIBC__)
```

Comment:
why?

Reply:
PTHREAD_MUTEX_RECURSIVE_NP is a non-portable glibc mutex and our docker base distro alpine does not have glibc only musl is available. On these systems only PTHREAD_MUTEX_RECURSIVE mutex is available that's why we redefine the non-portable symbol. An issue with a similar solution: https://github.com/godotengine/godot/issues/31555

Reply:
Could you add this answer to the top of the patch file? You can add random text above the diff command line, like in thirdparty/bzip2/bzip2.patch

Reply:
Good idea, added in 2192050b5ae17cbad8c4c700fb072e53c615c644